File: go/array/1184. 公交站间的距离.go
package array

func distanceBetweenBusStops(distance []int, start int, destination int) int {
	if start > destination {
		start, destination = destination, start
	}
	var total, res int
	for i := 0; i < len(distance); i++ {
		total += distance[i]
		if i >= start && i <= destination-1 {
			res += distance[i]
		}
	}
	if res < total-res {
		return res
	}
	return total - res
}

File: javascript/array/1184. 公交站间的距离.js

var distanceBetweenBusStops = function(distance, start, destination) {
    if (start > destination){
        let tmp = start
        start = destination
        destination = tmp
    }
    let total = 0;
    let res = 0;
    for (let i=0;i<distance.length;i++){
        total+=distance[i]
        if (i>=start && i<=destination-1){
            res+=distance[i]
        }
    }
    console.log(total,res)
    if (total-res > res){
        return res
    }
    return total - res
};